platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/ConfigurationAdmin.java

```diff
@@ -533,4 +553,325 @@ public void setFilterList(List<String> filterList) {
     void setMBeanServer(MBeanServer server) {
         mBeanServer = server;
     }
+
+    private static class CardinalityTransformer implements Transformer {
+        private final List<Map<String, Object>> metatype;
+
+        private final String pid;
+
+        public CardinalityTransformer(List<Map<String, Object>> metatype, String pid) {
+            this.metatype = metatype;
+            this.pid = pid;
+        }
+
+        @Override
+        // the method signature precludes a safer parameter type
+        @SuppressWarnings("unchecked")
+        public Object transform(Object input) {
+            Map.Entry<String, Object> entry = (Map.Entry<String, Object>) input;
```

Comment:
I'd do an instanceof check here to make sure that the input is what you expect and error out if it isn't.